### ISSUE TITLE
Working with InStream to Attach Files to E-Mail

### ISSUE THREAD

**Issue by CKL-CAalstede** [microsoft/ALAppExtensions#5176]
While converting our Solution to AL we noticed that restricting it to the App Store requirements is blocking us from working with Temporary Files while attaching Reports from Table 477 Report Inbox. 
The solution I am trying out right now is switching from attaching Files to Attaching InStream. Since all the necessary functionality can be found in Codeunit 400 SMTP Mail, I thought this won't be an issue, but right now I am running into problems and I can't find a way to make it work.

While the Reports do get attached with my current coding, for some reason, the very first one comes as an empty file, all others are perfect. Any advice or help with this implementation would be greatly appreciated.

Screenshot of my Coding( also attached as .txt):
![image](https://user-images.githubusercontent.com/52314835/70420878-f191b300-1a68-11ea-884f-f36d11f3b9ba.png)

If you replace our Recipient E-Mail Adress (Send Report Adress) with a text constant the coding would work in standard as well.
[COD50000.txt](https://github.com/microsoft/ALAppExtensions/files/3938639/COD50000.txt)

PS: Because we aim to keep a unitary code bases between the versions of our Solution I am trying this development in C/AL first and will convert to AL as soon as this works.

**Comment by AlexanderYakunin** [comment]
@JesperSchulz

**Comment by IhorHandziuk** [comment]
Hi @CKL-CAalstede , it looks like the problem is that you create an InStream before calling CalcFields. In order to have the data in a BLOB field, you need to call CalcFields on it first.
Hope this helps :)

**Comment by CKL-CAalstede** [comment]
I have tested you solution just now and still arrive at the same Issue. For some reason, the very last attachment comes up empty. 
Could you provide me wit any more feedback? 
Perhaps an Idea on how to restructure my Code?

I will soon rewrite my Code so it only uses Standard Fields so that you can test it in your Solution as well.

Best Regards 
Ciprian

**Comment by JesperSchulz** [comment]
Just for clarification: initially you write "the very first one comes as an empty file", now you write "the very last attachment comes up empty". Has the behavior changed? Or was this a typo?

**Comment by CKL-CAalstede** [comment]
The behavior has not changed I was wrong in my Assumption before. It is indeed the las Attachment that comes up empty.

**Comment by IhorHandziuk** [comment]
Nothing unexpected should happen when the AddAttachmentStream function is called. Can you, please, check if you don't have any records with an empty "Report Output" field in the "Report Inbox" table?

Otherwise, yes, please, send us a full example that we can repro on our end :)

**Comment by CKL-CAalstede** [comment]
Done it :).
Attached you will find the necessary objects and screenshots to reproduce my scenario. Everything I have used is based in Standard BC 14 functionality. I don't believe the Report I chose is of any relevancy but the same can be reproduce with any other Report.
[E-Mail Attachments of Report Inbox Pdfs.zip](https://github.com/microsoft/ALAppExtensions/files/4035959/E-Mail.Attachments.of.Report.Inbox.Pdfs.zip)

**Comment by JesperSchulz** [comment]
I've tried to run your code with Ihor - everything seems to be working just fine! We're a little at a loss as to why this isn't working on your end. Unfortunately we're going to have to close this issue down, as the problem doesn't seem to be lying on our end - or at least we cannot repro, which makes it very hard for us to assist you any further. Maybe you could ask in some of our larger forums, if anyone else in the community has experienced something similar? Here are the links to some of the larger forums, which you undoutbly know ;-):

[Business Central Community](https://community.dynamics.com/business/f/758)
[mibuso forum](https://forum.mibuso.com/categories/nav-three-tier)
[Dynamics User Group](https://dynamicsuser.net/nav/f/developers)

Sorry we couldn't solve your problem. Hopefully you find a solution with the help of our great community.

**Comment by CKL-CAalstede** [comment]
I have come to a solution!
It's still a mistery why my previous coding wasn't working but with the help of our great community I have finally come to a soultion :).
The change I took was how I parsed the Report Inbox Entries. I now parse the Code for every relevant Report Inbox Entry, while previously I parsed every Job Queue Entry and then jumped to the Report Inbox Entry (hope this makes sense).
